### PR TITLE
Fix bug where mousedown/touchstart inside the child, then mouseup/touchend outside the child doesn't reset the insideClick flag.

### PR DESCRIPTION
**Pull request by andysprout**
### Changes

reset the `clickInside` flag on window mouse up and window touch end.

### CODE REVIEW

**Comment by andysprout** [comment]
ping @nkbt

File: src/PageClick.js [hunk 1]
```diff
@@ -27,17 +27,21 @@ const PageClick = React.createClass({
...
 
   shouldComponentUpdate,
 
 
   componentWillUnmount() {
-    global.window.removeEventListener('mousedown', this.onDocumentClick, false);
     global.window.removeEventListener('touchstart', this.onDocumentClick, false);
+    global.window.removeEventListener('touchend', this.onDocumentMouseUp, false);
+    global.window.removeEventListener('mousedown', this.onDocumentClick, false);
+    global.window.removeEventListener('mouseup', this.onDocumentMouseUp, false);
```
Comment (nkbt) [inline src/PageClick.js]: With this change we do not need to re-define onMouseUp and onTouchEnd on element itself, so that code can be safely removed.

Comment (andysprout) [inline src/PageClick.js]: good point, fixed.

Comment (nkbt) [inline src/PageClick.js]: You can also work on `packetloop/repo` and do PR directly to the upstream.

Comment (nkbt) [inline src/PageClick.js]: There is not much sense to merge this PR to packetloop/repo/master really.

Comment (andysprout) [inline src/PageClick.js]: ok

**Pull request closed without merging**